### PR TITLE
planner: fix a bug that can't find column in indexLookupJoin with virtual generated column

### LINKED ISSUE #18515: IndexLookup Join can't find column with virtual generated column
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
mysql> create table t(a int, b json, c int AS (JSON_EXTRACT(b, '$.population')), key(c));
Query OK, 0 rows affected (0.01 sec)
mysql> desc select /*+ TIDB_INLJ(t2) */ t1.a, t1.c, t2.a from t t1, t t2 where t1.c=t2.c;
ERROR 1105 (HY000): Can't find column test.t.b in schema Column: [test.t.a,test.t.c] Unique key: []
```

### 2. What did you expect to see? (Required)
No error
### 3. What did you see instead (Required)
error
### 4. Affected version (Required)
master, 4.0

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: close #18515 

Problem Summary:

ExpandVirtualColumn is missed.

### What is changed and how it works?

1. let finishCopTask handle ExpandVirtualColumn.
2. Move `(p PointGetPlan) Init` to initialize.go

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

### Release note 

- fix a bug that can't find column in indexLookupJoin with virtual generated column

### CODE REVIEW

**Comment by bb7133** [comment]
```
 ==================
[2020-07-14T02:48:07.003Z] WARNING: DATA RACE
[2020-07-14T02:48:07.003Z] Write at 0x00c005674788 by goroutine 56:
[2020-07-14T02:48:07.003Z]   github.com/pingcap/tidb/executor.ResetContextOfStmt()
[2020-07-14T02:48:07.003Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/executor/executor.go:1666 +0xe1d
[2020-07-14T02:48:07.003Z]   github.com/pingcap/tidb/session.(*session).ExecuteStmt()
[2020-07-14T02:48:07.003Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/session/session.go:1140 +0x1f1
[2020-07-14T02:48:07.003Z]   github.com/pingcap/tidb/session.(*session).Execute()
[2020-07-14T02:48:07.003Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/session/session.go:1081 +0x3ee
[2020-07-14T02:48:07.003Z]   github.com/pingcap/tidb/session.execRestrictedSQL()
[2020-07-14T02:48:07.003Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/session/session.go:835 +0x15c
[2020-07-14T02:48:07.003Z]   github.com/pingcap/tidb/session.(*session).ExecRestrictedSQLWithContext()
[2020-07-14T02:48:07.003Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/session/session.go:774 +0x25b
[2020-07-14T02:48:07.003Z]   github.com/pingcap/tidb/session.(*session).ExecRestrictedSQL()
[2020-07-14T02:48:07.003Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/session/session.go:744 +0x92
[2020-07-14T02:48:07.003Z]   github.com/pingcap/tidb/statistics/handle.(*Handle).Update()
[2020-07-14T02:48:07.003Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/statistics/handle/handle.go:162 +0x1eb
[2020-07-14T02:48:07.003Z]   github.com/pingcap/tidb/domain.(*Domain).loadStatsWorker()
[2020-07-14T02:48:07.003Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/domain/domain.go:1096 +0x571
[2020-07-14T02:48:07.003Z] 
[2020-07-14T02:48:07.003Z] Previous read at 0x00c005674788 by goroutine 111:
[2020-07-14T02:48:07.003Z]   github.com/pingcap/tidb/executor.(*IndexLookUpExecutor).startIndexWorker.func1()
[2020-07-14T02:48:07.003Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/executor/distsql.go:514 +0x477
[2020-07-14T02:48:07.003Z] 
[2020-07-14T02:48:07.003Z] Goroutine 56 (running) created at:
[2020-07-14T02:48:07.003Z]   github.com/pingcap/tidb/domain.(*Domain).UpdateTableStatsLoop()
[2020-07-14T02:48:07.003Z]       /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/domain/domain.go:1041 +0x332
[2020-07-14T02:48:07.003Z]   github.com/pingcap/tidb/session.BootstrapSession()
```
Failed UT case

**Comment by qw4990** [comment]
PTAL @wjhuang2016  Please resolve conflicts.

File: planner/core/task.go [hunk 1]
```diff
@@ -711,6 +711,14 @@ func finishCopTask(ctx sessionctx.Context, task task) task {
 	if !ok {
 		return task
 	}
+	if t.tablePlan != nil {
+		tp := t.tablePlan
+		for len(tp.Children()) > 0 {
+			tp = tp.Children()[0]
```
Comment (lzmhhh123) [inline planner/core/task.go:717]:
```suggestion
			if len(tp.Children()) == 1 {
				tp = tp.Children()[0]
			} else {
				join := tp.(*PhysicalBroadCastJoin)
				tp = join.children[1-join.InnerChildIdx]
			}
```

We should consider broadcast join in cop task. And by the way, this check can be moved to https://github.com/pingcap/tidb/pull/18516/files#diff-2f445d8cb9c2f7d8ec4fb0586ae1f510R730

**Pull request merged**